graphbolt/src/fused_csc_sampling_graph.cc

```diff
@@ -484,6 +495,299 @@ auto GetTemporalPickFn(
+                              : 0;
+                      const auto seed_index =
+                          (hetero_with_seed_offsets)
+                              ? i - seed_offsets->at(seed_type_id)
+                              : i;
+
+                      // Step 4. Pick neighbors for each node.
+                      picked_number = pick_fn(
+                          offset, num_neighbors, picked_eids_data_ptr,
+                          seed_index, subgraph_indptr_data_ptr,
+                          etype_id_to_num_picked_offset);
+                      if (!hetero_with_seed_offsets) {
+                        TORCH_CHECK(
+                            num_picked_neighbors_data_ptr[i + 1] ==
+                                picked_number,
+                            "Actual picked count doesn't match the calculated "
+                            "pick number.");
+                      }
+
+                      // Step 5. Calculate other attributes and return the
+                      // subgraph.
+                      if (picked_number > 0) {
+                        AT_DISPATCH_INDEX_TYPES(
+                            subgraph_indices.scalar_type(),
+                            "IndexSelectSubgraphIndices", ([&] {
+                              auto subgraph_indices_data_ptr =
+                                  subgraph_indices.data_ptr<index_t>();
+                              auto indices_data_ptr =
+                                  indices_.data_ptr<index_t>();
+                              for (auto i = 0; i < num_etypes; ++i) {
+                                if (etype_id_to_dst_ntype_id[i] != seed_type_id)
+                                  continue;
+                                const auto indptr_offset =
+                                    with_seed_offsets
+                                        ? etype_id_to_num_picked_offset[i] +
+                                              seed_index
+                                        : seed_index;
+                                const auto picked_begin =
+                                    subgraph_indptr_data_ptr[indptr_offset];
+                                const auto picked_end =
+                                    subgraph_indptr_data_ptr[indptr_offset + 1];
+                                for (auto j = picked_begin; j < picked_end;
+                                     ++j) {
+                                  subgraph_indices_data_ptr[j] =
+                                      indices_data_ptr[picked_eids_data_ptr[j]];
+                                  if (hetero_with_seed_offsets &&
+                                      node_type_offset_.has_value())
+                                    // Substract the node type offset from
+                                    // subgraph indices.
+                                    AT_DISPATCH_INDEX_TYPES(
+                                        node_type_offset_.value().scalar_type(),
```

Comment:
do we need a dispatch here? Doesn't the node_type_offset match the indices dtype already?

Reply:
Is this always the case? I wasn't sure about that so I added a dispatch.

Reply:
You can try removing it and see if all the tests pass locally rather quickly. I run graphbolt tests only for quick testing before pushing code.